runtime/commonMain/src/kotlinx/serialization/PlatformUtils.kt

```diff
@@ -7,9 +7,23 @@ package kotlinx.serialization
 import kotlinx.serialization.internal.defaultSerializer
 import kotlin.reflect.KClass
 
+/**
+ * Retrieves a [KSerializer] for the given [KClass].
+ *
+ * Class must be annotated as @Serializable or be one of the built-in types.
+ * Class must have no generic parameters.
+ *
+ * This method may sometimes fail, see [ImplicitReflectionSerializer] documentation.
+ *
+ * To work with generic classes, prefer using `serializer(KType)` function.
```

Comment:
"Prefer" is an incorrect recommendation here. 
`For classes with type parameters please use 'serializer(KType)' instead`